sympy/solvers/tests/test_inequalities.py

```diff
@@ -318,6 +318,18 @@ def test_issue_8974():
     assert isolve(oo > x, x) == And(-oo < x, x < oo)
 
 
+def test_issue_10198():
+    assert reduce_inequalities(
+        -1 + 1/abs(1/x - 1) < 0) == Or(
+        And(-oo < x, x < 0), And(S(0) < x, x < S(1)/2)
+        )
+    assert reduce_inequalities(abs(1/sqrt(x)) - 1, x) == Eq(x, -1)
```

Comment:
I guess we are checking for `n.integer`, so this test case should not be required.